clang/lib/Sema/SemaExpr.cpp

```diff
@@ -212,10 +212,17 @@ bool Sema::DiagnoseUseOfDecl(NamedDecl *D, ArrayRef<SourceLocation> Locs,
                              ObjCInterfaceDecl *ClassReceiver) {
   if (getLangOpts().SYCLIsDevice) {
     if (auto VD = dyn_cast<VarDecl>(D)) {
+      bool IsConst = VD->getType().isConstant(Context);
+      if (VD->getTLSKind() != VarDecl::TLS_None)
+        SYCLDiagIfDeviceCode(*Locs.begin(), diag::err_thread_unsupported);
+
       if (VD->getStorageClass() == SC_Static &&
```

Comment:
Since both these diagnostics are about the !IsConst cases, I have a small preference if this were refactored to check that first.   Would save checking the other conditions for the const cases.